ci-operator/step-registry/baremetalds/e2e/upgrade/upgrade/baremetalds-e2e-upgrade-upgrade-commands.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
```

Comment:
This file is not required - `openshift-e2e-test` handles that

Reply:
It is not possible to use the `openshift-e2e-test` in the current workflow since we're using a different cluster profile (`packet`) and dev-scripts to setup remotely the cluster (see `baremetalds-devscript-setup` step). @eisraeli I think anyhow that it could be worth reusing the same approach in `baremetalds-e2e-test` step, ie running upgrade tests if a variable is set, in order to have a single test step for the baremetal workflow.

Reply:
`openshift-e2e-tests` reads KUBECONFIG and doesn't need direct access to the cluster

Reply:
I don't think the current implementation support `packet` profile (see https://github.com/openshift/release/blob/1d793ed9a62b52108d855a5d7145956c7271c3c7/ci-operator/step-registry/openshift/e2e/test/openshift-e2e-test-commands.sh#L51), and currently our KUBECONFIG points to a file generated by `dev-scripts` in the remote packet instance (https://github.com/openshift/release/blob/1d793ed9a62b52108d855a5d7145956c7271c3c7/ci-operator/step-registry/baremetalds/devscripts/setup/baremetalds-devscripts-setup-commands.sh#L99). Moreover, I would not recommend such refactoring within the scope of the current PR.

Reply:
You'd have to add support for it then. I don't think it has any specific flags, you'd need to reuse existing legacy templates.

>currently our KUBECONFIG points to a file generated by dev-scripts in the remote packet instance

Copy it to `${SHARED_DIR}` - AWS install does that:
```
cp \
    -t "${SHARED_DIR}" \
    "${dir}/auth/kubeconfig" \
    "${dir}/metadata.json"
```

>Moreover, I would not recommend such refactoring within the scope of the current PR.

This is necessary for this PR to land and be approved by OTA team

Reply:
We already are supporting the `packet` cluster profile from the very beginning, and the current baremetal workflow is the one that is managing it (see also the dedicated packet steps)

Reply:
I should mention, it’s not as simple as a kubeconfig. The API server is not accessible remotely. We do things like support IPv6 which has to run inside the packet host where we emulate bare metal. 

What is the OTA team?

Reply:
> The API server is not accessible remotely

I'm not seeing any tricks to access it in this commit. If `packet` requires any I suppose we can fork the upgrade test step, but that is very undesirable and goes against the whole purpose of multistep testing.

>What is the OTA team?

Team taking care of upgrades, including CI upgrade tests.

Reply:
>>  The API server is not accessible remotely
>
> I'm not seeing any tricks to access it in this commit. If packet requires any I suppose we can fork the upgrade test step, but that is very undesirable and goes against the whole purpose of multistep testing.

What do you mean? It's not about Packet. It's because we save significant amounts of money by running baremetal clusters through emulated baremetal with libvirt + vbmc/sushy-tools. The cluster is contained within the libvirt networks on the host.

The entire purpose of baremetalds-e2e-upgrade-test-commands is that it runs the e2e-tests suites by SSH'ing to the host.  I agree about mulisteps tests should be useful for everyone, and in fact the baremetalds workflows existed long before any other platforms did, and when it came time to create the ipi, upgrade, etc tests our specific use cases weren't considered so most of our steps remained special.

Reply:
>The cluster is contained within the libvirt networks on the host, therefore the API is not exposed on the public IP

This workflow would create pods on api.ci / build01 / build02 CI cluster. How would they reach the created cluster?
 
This should should not hardcode the release, it should not use `oc adm upgrade` or any other home-grown way to upgrade the cluster. It should use `openshift-tests run-upgrade` command to upgrade from `OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE` to `OPENSHIFT_UPGRADE_RELEASE_IMAGE_OVERRIDE`

Reply:
- They do not reach directly the cluster, all the operations are ssh'ed to the target Packet host that is currently containing the newly spawned cluster
- We are in sync to not hardcode the release: when Eran started the activity the feature was not yet available for the multistage tests (the PR is still a draft)
- The goal is to run an `oc adm upgrade` - or any other standard test command that will perform the upgrade on the target cluster (ie like the `openshift-tests run-upgrade` you suggested I suppose)

Reply:
Ah, okay, that was my concern - upgrade should happen via `openshift-tests run-upgrade`. This test should be able to reach the target host via ssh (iiuc its already implemented in templates)